diff --git a/src/corelib/global/qendian.h b/src/corelib/global/qendian.h
index 257efbbd..9463b5be 100644
--- a/src/corelib/global/qendian.h
+++ b/src/corelib/global/qendian.h
@@ -48,6 +48,9 @@
 #include <stdlib.h>
 #include <string.h>
 
+// include limits for gcc 11
+#include <limits>
+
 #ifdef min // MSVC
 #undef min
 #undef max
diff --git a/src/corelib/global/qfloat16.h b/src/corelib/global/qfloat16.h
index c7a9c87a..b0a1abe8 100644
--- a/src/corelib/global/qfloat16.h
+++ b/src/corelib/global/qfloat16.h
@@ -45,6 +45,9 @@
 #include <QtCore/qmetatype.h>
 #include <string.h>
 
+// Add limits for GCC 11
+#include <limits>
+
 #if defined(QT_COMPILER_SUPPORTS_F16C) && defined(__AVX2__) && !defined(__F16C__)
 // All processors that support AVX2 do support F16C too. That doesn't mean
 // we're allowed to use the intrinsics directly, so we'll do it only for
diff --git a/src/corelib/text/qbytearray.h b/src/corelib/text/qbytearray.h
index f4c335ca..95c6b40f 100644
--- a/src/corelib/text/qbytearray.h
+++ b/src/corelib/text/qbytearray.h
@@ -52,6 +52,9 @@
 #include <string>
 #include <iterator>
 
+// Add limits for gcc 11
+#include <limits>
+
 #ifdef truncate
 #error qbytearray.h must be included before any header file that defines truncate
 #endif
